litestar/plugins.py

```diff
@@ -151,10 +152,27 @@ def to_openapi_schema(self, field_definition: FieldDefinition, schema_creator: S
         raise NotImplementedError()
 
 
+class OpenAPISchemaPlugin(OpenAPISchemaPluginProtocol):
```

Comment:
I added a new plugin class here, since adding these methods to the protocol would be breaking. I want to open up a new PR that adds abc's for all of these instead of the protocols, as it makes extension of the plugin interface in a backwards compatible way much easier.

Thoughts @litestar-org/maintainers?

Reply:
asked probably dumb question in DMs, otherwise i think its good